src/benchmark.cc

```diff
@@ -265,8 +265,9 @@ void RunBenchmarks(const std::vector<BenchmarkInstance>& benchmarks,
   BenchmarkReporter::Context context;
   context.name_field_width = name_field_width;
 
-  // Keep track of running times of all instances of current benchmark
-  std::vector<BenchmarkReporter::Run> complexity_reports;
+  // Keep track of running times of all instances of each benchmark family.
+  std::map<int /*family_index*/, std::vector<BenchmarkReporter::Run>>
```

Comment:
if we're worried about performance, this could be a `std::vector<std::vector<BenchmarkReporter::Run>>` where the family index is the index into the outer vector.

Reply:
Better yet, i don't create entries into this map unless we will need them, see code below.

Reply:
yeah, map access is still _much_ slower than vector index lookup. not going to block this CL on it though as there's no proof it's slowing anything down.